internal/credspropagation/azure.go

```diff
@@ -94,3 +96,15 @@ func generateAzureCCMSecret(azureCluster *capz.AzureCluster, azureClIdty *capz.A
 
 	return makeSecret("azure-cloud-provider", metav1.NamespaceSystem, secretData), nil
 }
+
+func getAzureSubnetData(azureCluster *capz.AzureCluster) (subnetName, secGroup, routeTable string) {
+	subnetName = azureCluster.Name + "-node-subnet"
```

Comment:
This logic will apply to both hosted and standalone Azure clusters. But for the hosted case, the subnet name is different. Would it be better to retrieve the subnet name directly from the AzureCluster spec by identifying the subnet with the `node` role?
Or, the bug does not affect hosted clusters and we may completely ignore the hosted case?

Reply:
Yep you're rignt. We're passing different subnet explicitly to hosted...
That was a really dumb mistake from my side, thanks for noticing!